doc/developer-notes.md

```diff
@@ -1390,7 +1391,7 @@ communication:
   ```
 
 - For consistency and friendliness to code generation tools, interface method
-  input and inout parameters should be ordered first and output parameters
+  input and in-out parameters should be ordered first and output parameters
   should come last.
```

Comment:
Maybe we can add something that we prefer returning values directly to "output parameters". This was written before the days of `std::optional` etc.

Reply:
Good idea, added 968a67b9be5262228b47184b9821fcab as a first attempt.